src/Workspaces/Core/Portable/Workspace/Host/TemporaryStorage/TrivialTemporaryStorageService.cs

```diff
@@ -94,10 +93,8 @@ public Task<SourceText> ReadTextAsync(CancellationToken cancellationToken = defa
 
             public void WriteText(SourceText text, CancellationToken cancellationToken = default(CancellationToken))
             {
-                // Decompose the SourceText into it's underlying parts, since we use it as a key
-                // into many other caches that don't expect it to be held
-                _text = text.ToString();
-                _encoding = text.Encoding;
+                // This is a trivial implementation, indeed.
+                _sourceText = text;
```

Comment:
I was thinking this might be an appropriate time to use object.MemberwiseClone()
It's more in the spirit of the original (create a copy of the SourceText so the original can be released) and wouldn't require a change for unit tests.
Thoughts?